File: Pharmacy/Entities/Product.cs
namespace Pharmacy.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Descripton { get; set; }
        public float Price { get; set; }
        public string PictureUrl { get; set; }
        public string Category { get; set; }
        public string Brand { get; set; }
        public int QuantityInStock { get; set; }
    }
}

File: client/src/app/models/product.ts
export interface Product {
    id: number;
    name: string;
    description?: string;
    price?: number;
    pictureUrl?: string;
    category?: string;
    brand?: string;
    quantityInStock?: number;
}

export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    brands: string[];
    categories: string[];
    pageNumber: Number;
    pageSize: Number;
}